Core/clim-basic/geometry/region-utilities.lisp

```diff
@@ -0,0 +1,1204 @@
+                                     (when done?
+                                       ;; iterate more nonetheless here -- is
+                                       ;; this a good idea?
+                                       (setf sol
+                                             (mapcar (lambda (x)
+                                                       (newton-ziel-gerade
+                                                        polynom x))
+                                                     sol))
+                                       (setf res (nconc sol res))
+                                       (return))))
+                                 (setf x1 x-start)
+                                 (setq n 0))))))
+                 (setf x (min 1d0 (max -1d0 x1)))        ;Is this allowed?
+                 (incf n)))
+             res)))))
+
+(defun horner-schema (polynom x)
+  ;; Evaluates the polynomial `polynom' by means of horner's method at the
+  ;; place `x'; returns two values:
+  ;; - the value of the function
+  ;; - the last line of horner's method (result of division)
+  (let ((n (length polynom)))
+    (cond ((= n 0) (values 0))
+          ((= n 1) (values (aref polynom 0) '#()))
+          (t
+           (let ((b (make-array (1- n))))
+             (setf (aref b 0) (aref polynom 0))
+             (do ((i 1 (+ i 1)))
+                 ((= i (- n 1))
+                  (values
+                   (+ (* (aref b (- i 1)) x) (aref polynom i))
+                   b))
+               (setf (aref b i) (+ (* (aref b (- i 1)) x) (aref polynom i)))))))))
+
+
+;;; routines for approximating ellipses as bezier curves
+
+;;;
+;;; Many backends, such as PDF and PostScript don't provide shape
+;;; drawing functions per se, but rather primitives for working with
+;;; paths such as lines and bezier curves. One can closely approximate
+;;; arbitrary ellipse path with appropriate bezier curves. A good
+;;; primer on drawing ellipses with lines, or quadratic or cubic
+;;; bezier curves can be found here:
+;;;
+;;; <https://www.spaceroots.org/documents/ellipse/index.html>.
+;;;
+;;; We use the algorithm described in the above reference to construct
+;;; cubic bezier curves.
+
+;;; I doubt that these functions belong in geometry utilities but I
+;;; move them anyway. -- jd 2019-10-10
+
+;;;
+;;; first some math utlitily functions
+(defun square (x)
+  "Returns the number X squared."
+  (* x x))
+
+(defun normalize-angle (angle)
+  "Takes an angle ANGLE and returns the corresponding non-negative angle
+less than or equal to 2pi. Note that 4pi would be normalized to 0, not
+2pi, but 2pi is normalized to 2pi."
+  (if (or (< angle 0)
+          (> angle (* pi 2)))
+      (mod angle (* pi 2))
+      angle))
+
+(defun find-angle* (x1 y1 x2 y2)
+  "Returns the angle between two vectors described by x1, y1 and x2,
+y2."
+  (let ((theta (- (phase (complex y2 x2))
+                  (phase (complex y1 x1)))))
+    (normalize-angle theta)))
```

Comment:
These should probably be in the "point and angle utilities" section.

Reply:
fixed